src/legendhpges/base.py

```diff
@@ -105,6 +111,107 @@ def _decode_polycone_coord(self) -> tuple[list[float], list[float]]:
         Must be overloaded by derived classes.
         """
 
+    def get_profile(self) -> tuple[list[float], list[float]]:
+        """Get the profile of the HPGe detector.
+
+        Returns
+        -------
+        (r.z)
+            two lists of r and z coordinates, respectively.
+
+        Note
+        -----
+            For V02160A and P00664A the detector profile is that of the solid without cut.
+        """
+        if isinstance(self.solid, geant4.solid.GenericPolycone) is False:
```

Comment:
```suggestion
        if not isinstance(self.solid, geant4.solid.GenericPolycone):
```

but here: should we add a sanity check that .obj1 is actually a polycone?